src/window-rules.c

```diff
@@ -13,20 +14,44 @@
 #include "view.h"
 #include "window-rules.h"
 
-void
-window_rules_apply(struct view *view, enum window_rule_event event)
+/* Try to match against identifier AND title (if set) */
+static bool
+view_matches_criteria(struct window_rule *rule, struct view *view)
 {
-	const char *identifier = view_get_string_prop(view, "app_id");
-	if (!identifier) {
-		return;
+	const char *id = view_get_string_prop(view, "app_id");
+	const char *title = view_get_string_prop(view, "title");
+
+	if (rule->identifier && rule->title) {
+		if (!id || !title) {
+			return false;
+		}
+		return match_glob(rule->identifier, id)
+			&& match_glob(rule->title, title);
+	} else if (rule->identifier) {
+		if (!id) {
+			return false;
+		}
+		return match_glob(rule->identifier, id);
+	} else if (rule->title) {
+		if (!title) {
+			return false;
+		}
+		return match_glob(rule->title, title);
+	} else {
+		wlr_log(WLR_ERROR, "rule has no identifier or title\n");
```

Comment:
Another candidate for a `validate()` check in `src/rcxml.c`

Reply:
Have added `validate()` which removes window-rules without criteria.